ephios/plugins/questionnaires/models.py

```diff
@@ -0,0 +1,120 @@
+        verbose_name = _("Question")
+        verbose_name_plural = _("Questions")
+
+    def __str__(self):
+        # pylint: disable=invalid-str-returned
+        return self.name
+
+    def get_signup_formfield(
+        self, participant: AbstractParticipant, participation: AbstractParticipation, signup_choice
+    ):
+        # Restore answer from participation (when editing) or the user's saved answers (if local participant)
+        answer_key = str(self.pk)
+        if answer_key in participation.questionnaire_answers:
+            initial = participation.questionnaire_answers[answer_key]
+        elif (
+            isinstance(participant, LocalUserParticipant)
+            and answer_key in participant.user.saved_questionnaire_answers
+        ):
+            initial = participant.user.saved_questionnaire_answers[answer_key]
+        else:
+            initial = None
+
+        # Reset answer if selected option does not exist
+        if self.type != self.TYPE_TEXT and initial not in self.choices:
+            initial = None
+
+        field_name = self.get_form_slug()
+        required = self.required if initial is not None else False
+
+        form_class = None
+        form_kwargs = {
+            "label": self.question_text,
+            "help_text": self.description,
+            "initial": initial,
+            "required": required,
+        }
+
+        serializer_class = None
+        serializer_kwargs = {
+            "required": required,
+        }
+
+        if self.type == self.TYPE_TEXT:
+            form_class = forms.CharField
+            serializer_class = serializers.CharField
+        elif self.type == self.TYPE_SINGLE_RADIO:
+            form_class = forms.ChoiceField
+            serializer_class = serializers.ChoiceField
+            form_kwargs["widget"] = forms.RadioSelect
+        elif self.type == self.TYPE_SINGLE_LIST:
+            form_class = forms.ChoiceField
+            serializer_class = serializers.ChoiceField
+            form_kwargs["widget"] = Select2Widget
+        elif self.type == self.TYPE_MULTIPLE:
+            form_class = forms.MultipleChoiceField
+            serializer_class = serializers.MultipleChoiceField
+            form_kwargs["widget"] = Select2MultipleWidget
+
+        if self.type != self.TYPE_TEXT:
+            assert isinstance(
+                self.choices, list
+            ), f"The choices of question {self.name} are not a list"
+            # pylint: disable=not-an-iterable
+            choices = [(choice, choice) for choice in self.choices]
+            form_kwargs["choices"] = choices
+            serializer_kwargs["choices"] = choices
+
+        return AdditionalField(
+            field_name, form_class, form_kwargs, serializer_class, serializer_kwargs
+        )
+
+    def get_form_slug(self):
+        return slugify(f"{self.pk} {self.name}")
+
+    @staticmethod
+    def get_pk_from_slug(slug: str):
+        return slug.split(".", maxsplit=1)[1].split("-")[0]
```

Comment:
Please add a docstring with an example.

Reply:
Added in 8fa586a3fadb430fd1493e34af729970ea4b73cf